model-client-js-test/replicated-model-test/src/ReplicatedModelJS.test.ts

```diff
@@ -0,0 +1,43 @@
+import { org } from "@modelix/model-client";
+import { randomUUID } from "crypto";
+
+const MODEL_SERVER_URL = "http://localhost:28103/v2"
+
+const connectClient = org.modelix.model.client2.connectClient
+type ClientJS = org.modelix.model.client2.ClientJS
+type ReplicatedModelJS = org.modelix.model.client2.ReplicatedModelJS
+
+test("replicated model uses user set in client", async () => testWithEmptyReplicatedModel(async (client, replicatedModel) => {
+  client.setClientProvideUserId("aAuthor")
+  const rootNode = replicatedModel.getBranch().rootNode
+
+  rootNode.setPropertyValue("aPropety", "aValue")
+
+  const versionInformation = await replicatedModel.getCurrentVersionInformation()
+  const lastAutor = versionInformation.author
+  expect(lastAutor).toBe("aAuthor");
+}));
+
+test("replicated model repturn user set in client", async () => testWithEmptyReplicatedModel(async (client, replicatedModel) => {
+  const versionInformation = await replicatedModel.getCurrentVersionInformation()
+  const lastTimestamp = versionInformation.time
+  expect(lastTimestamp).toBeTruthy()
+}));
+
+async function testWithEmptyReplicatedModel( testBlock: (client: ClientJS, replicatedModel: ReplicatedModelJS) => Promise<void>) {
```

Comment:
@drik98 Do you know some technique in TS/Jest to make this test easier to read?

Reply:
I think it would already be easier if you just await the result instead of using a callback. 

```typescript
async function generateEmptyReplicatedModel(): { clientJs: ClientJS, replicatedModel: ReplicatedModelJS};

test("replicated model repturn user set in client", async () => {
  const {replicatedModel} = await generateEmptyReplicatedModel();
  const versionInformation = await replicatedModel.getCurrentVersionInformation()
  const lastTimestamp = versionInformation.time
  expect(lastTimestamp).toBeTruthy()
});
```

Then combine that with a beforeEach/afterEach hook to add the disposing https://jestjs.io/docs/setup-teardown 

```typescript
let client: ClientJS | undefined;
let replicatedModel: ReplicatedModelJS | undefined;

beforeEach(() => {
   ({client, replicatedModel}) = generateEmptyReplicatedModel();
})

afterEach(() => {
    dispose(); // or something
})


test("replicated model repturn user set in client", async () => {
  const versionInformation = await replicatedModel.getCurrentVersionInformation()
  const lastTimestamp = versionInformation.time
  expect(lastTimestamp).toBeTruthy()
});
```

Reply:
thx